pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,651 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the MPLDrawer class for creating circuit diagrams with matplotlib
+"""
+from collections.abc import Iterable
+
+import matplotlib.pyplot as plt
+from matplotlib import patches
+
+
+def _to_tuple(a):
+    """converts int or iterable to always tuple"""
+    if a is None:
+        return tuple()
+    if isinstance(a, Iterable):
+        return tuple(a)
+    return (a,)
+
+
+class MPLDrawer:
+    r"""Allows easy creation of graphics representing circuits with Matplotlib
+
+    Args:
+        n_layers (Int): the number of layers
+        n_wires (Int): the number of wires
+
+    Keyword Args:
+        wire_kwargs=None (dict): matplotlib configuration options for drawing the wire lines
+        figsize=None (Iterable): Allows users to manually specify the size of the figure.  Defaults
+            to scale with the size of the circuit via ``n_layers`` and ``n_wires``.
+
+    **Example**
+
+    .. code-block:: python
+
+        def example():
```

Comment:
I'd probably remove the `def` and just run the example without it. There are also a few minor things I'd think worth changing, to make it a bit clearer:
* adding spaces after commas
* (as @glassnotes mentioned above) use the layers/wires keywords in all drawer function calls

Reply:
I wrapped it in a function so we can call the same thing with a stylesheet and edited rcParams and see how things change. I could either just include all that code again.